crates/turborepo-ffi/src/lib.rs

```diff
@@ -121,3 +122,26 @@ pub extern "C" fn previous_content(buffer: Buffer) -> Buffer {
     };
     resp.into()
 }
+
+#[no_mangle]
+pub extern "C" fn recursive_copy(buffer: Buffer) -> Buffer {
+    let req: proto::RecursiveCopyRequest = match buffer.into_proto() {
+        Ok(req) => req,
+        Err(err) => {
+            let resp = proto::RecursiveCopyResponse {
+                error: Some(err.to_string()),
+            };
+            return resp.into();
+        }
+    };
+    let response = match turborepo_fs::recursive_copy(
+        &AbsoluteSystemPathBuf::new_unchecked(req.src),
+        &AbsoluteSystemPathBuf::new_unchecked(req.dst),
```

Comment:
Since `RecursiveCopy` is only used in prune and that isn't exactly performance sensitive, I'd rather skip the addition of `new_unchecked` and either unwrap the `Result` type from `new` or something like `map_err(|e| e.to_string()`. I'm just worried  of possible `new_unchecked`

Reply:
I'm making the change for this and removing `new_unchecked`. It's possible that it will be back at some point, in the Go version we have `AnchoredSystemPathFromUpstream` for places where we know some API returns an absolute path. But the FFI code is (hopefully) short-lived and (hopefully) not in a hot loop, so I think it's fine to do the check.